### ISSUE TITLE
which weight can be used to generate a pseudo-rain image

### ISSUE THREAD

**Issue by Twq521** [GuangluDong0728/CSUD-Unsupervised-Deraining-CVPR2025#4]
Hello author, which weight can be used to generate a pseudo-rain image?

<img width="1223" height="340" alt="Image" src="https://github.com/user-attachments/assets/aa8875d5-26fb-45e0-83ec-cf23f3c70011" />

**Comment by Twq521** [comment]
I would be very grateful if the author could take a moment out of your busy schedule to answer my question. It's really important to me.

**Comment by GuangluDong0728** [comment]
Hello, thank you for showing interest in our work. Recently, I’ve been quite busy with moving house. I can send you the weight of the generator via email in two days. For the past two days, I haven’t had access to a computer. I somewhat forgot whether I had provided the weight for the generator at github. However, the network structure of the generator has definitely been given at github. If you’re in a hurry, you can try training it first. It should only take less than a day to complete. Alternatively, you can wait for me for two more days. Thank you for your understanding.



---- Replied Message ----
| From | ***@***.***> |
| Date | 06/17/2026 23:31 |
| To | ***@***.***> |
| Cc | ***@***.***> |
| Subject | Re: [GuangluDong0728/CSUD-Unsupervised-Deraining-CVPR2025] which weight can be used to generate a pseudo-rain image (Issue #4) |
Twq521 left a comment (GuangluDong0728/CSUD-Unsupervised-Deraining-CVPR2025#4)

I would be very grateful if the author could take a moment out of your busy schedule to answer my question. It's really important to me.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications, keep track of coding agent tasks and review pull requests on the go with GitHub Mobile for iOS and Android. Download it today!

**Comment by Twq521** [comment]
Hello author, when would be a convenient time for you to upload the generator weights? I need to use them to synthesize some rain images. Thank you. If it's really not possible, I'll try training it myself.

**Comment by Twq521** [comment]
> Hello, thank you for showing interest in our work. Recently, I’ve been quite busy with moving house. I can send you the weight of the generator via email in two days. For the past two days, I haven’t had access to a computer. I somewhat forgot whether I had provided the weight for the generator at github. However, the network structure of the generator has definitely been given at github. If you’re in a hurry, you can try training it first. It should only take less than a day to complete. Alternatively, you can wait for me for two more days. Thank you for your understanding.
> […](#)

Hi，author，I would be very grateful if you can take a moment out of your busy schedule to provide the weight for the generator. It's really important to me.

**Comment by GuangluDong0728** [comment]
I sincerely apologize for the late reply — things have been extremely busy on my end recently.

I have uploaded the generator weights from our framework to Google Drive. Specifically, I have uploaded the generator weights trained on two datasets: Rain100L and RealRain1K-L. The links are as follows:

Rain100L: https://drive.google.com/file/d/1ZvtoaFyxGorWHavdx2HanfSqThmlDXdZ/view?usp=drive_link
RealRain1K-L: https://drive.google.com/file/d/1enUuRnxGQ2ipzbcigi_Qcf_jpjJuRUTi/view?usp=drive_link
The generator in our framework takes two inputs: a clean image and an unpaired rainy image. It synthesizes a rainy image on the clean image such that the rain streak distribution resembles that of the input rainy image.

Since the generator needs to be trained separately for each dataset, different weights are required to synthesize rainy images for different datasets. For now, I have only uploaded the weights for Rain100L and RealRain1K-L. Could you let me know which specific datasets you need? If you require additional ones, I am happy to upload them as well.

I hope this has not caused any delays to your work. Best wishes and good luck with your research!



At 2026-07-08 10:41:19, "Twq521" ***@***.***> wrote:

Twq521 left a comment (GuangluDong0728/CSUD-Unsupervised-Deraining-CVPR2025#4)

Hello, thank you for showing interest in our work. Recently, I’ve been quite busy with moving house. I can send you the weight of the generator via email in two days. For the past two days, I haven’t had access to a computer. I somewhat forgot whether I had provided the weight for the generator at github. However, the network structure of the generator has definitely been given at github. If you’re in a hurry, you can try training it first. It should only take less than a day to complete. Alternatively, you can wait for me for two more days. Thank you for your understanding.
…

Hi，author，I would be very grateful if you can take a moment out of your busy schedule to provide the weight for the generator. It's really important to me.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications, keep track of coding agent tasks and review pull requests on the go with GitHub Mobile for iOS and Android. Download it today!

**Comment by Twq521** [comment]
> I sincerely apologize for the late reply — things have been extremely busy on my end recently.
> 
> I have uploaded the generator weights from our framework to Google Drive. Specifically, I have uploaded the generator weights trained on two datasets: Rain100L and RealRain1K-L. The links are as follows:
> 
> Rain100L: https://drive.google.com/file/d/1ZvtoaFyxGorWHavdx2HanfSqThmlDXdZ/view?usp=drive_link
> RealRain1K-L: https://drive.google.com/file/d/1enUuRnxGQ2ipzbcigi_Qcf_jpjJuRUTi/view?usp=drive_link
> The generator in our framework takes two inputs: a clean image and an unpaired rainy image. It synthesizes a rainy image on the clean image such that the rain streak distribution resembles that of the input rainy image.
> 
> Since the generator needs to be trained separately for each dataset, different weights are required to synthesize rainy images for different datasets. For now, I have only uploaded the weights for Rain100L and RealRain1K-L. Could you let me know which specific datasets you need? If you require additional ones, I am happy to upload them as well.
> 
> I hope this has not caused any delays to your work. Best wishes and good luck with your research!
> 
> 
> 
> At 2026-07-08 10:41:19, "Twq521" ***@***.***> wrote:
> 
> Twq521 left a comment ([GuangluDong0728/CSUD-Unsupervised-Deraining-CVPR2025#4](https://github.com/GuangluDong0728/CSUD-Unsupervised-Deraining-CVPR2025/issues/4))
> 
> Hello, thank you for showing interest in our work. Recently, I’ve been quite busy with moving house. I can send you the weight of the generator via email in two days. For the past two days, I haven’t had access to a computer. I somewhat forgot whether I had provided the weight for the generator at github. However, the network structure of the generator has definitely been given at github. If you’re in a hurry, you can try training it first. It should only take less than a day to complete. Alternatively, you can wait for me for two more days. Thank you for your understanding.
> …
> 
> Hi，author，I would be very grateful if you can take a moment out of your busy schedule to provide the weight for the generator. It's really important to me.
> 
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> Triage notifications, keep track of coding agent tasks and review pull requests on the go with GitHub Mobile for iOS and Android. Download it today!
> You are receiving this because you commented.Message ID: ***@***.***>

Thank you so much! The generator weights you uploaded have been incredibly helpful to me. However, I have two more questions I’d like to ask:
1. Opinions online regarding the Rain100L/H version are mixed, and so far, I’ve only managed to find the test set for Rain100L/H. Could you please share the training set or let me know where I can download it?
2. Would it be possible for you to upload the generator weights for Rain100H and RealRain1K-H as well? I need those too.
